contributors/design-proposals/auditing.md

```diff
@@ -0,0 +1,276 @@
+
+The log output format of https://github.com/kubernetes/kubernetes/pull/27087 is fixed. It is text based, unstructured (e.g. non-JSON) data which must be parsed to be usable in any advanced external system used to analyze audit logs.
+
+The log output format does not follow any public standard like e.g. https://www.dmtf.org/standards/cadf.
+
+With this proposal we describe how the auditing functionality can be extended in order:
+
+- to allow multiple output formats, e.g. access.log style or structured JSON output
+- to allow deep payload inspection to allow
+  - either real differential JSON output (which field of an object have changed from which value to which value)
+  - or full object output of the new state (and optionally the old state)
+- to be extensible in the future to fully comply with the Cloud Auditing Data Federation standard (https://www.dmtf.org/standards/cadf)
+- to allow filtering of the output
+  - by kind, e.g. don't log endpoint objects
+  - by object path (JSON path), e.g. to ignore all `*.status` changes
+  - by user, e.g. to only log end user action, not those of the controller-manager and scheduler
+  - by level (request headers, request object, storage object)
+
+while
+
+- not degrading apiserver performance when auditing is disabled.
+
+## Constraints and Assumptions
+
+* it is not the goal to implement all output formats one can imagine. The  main goal is to be extensible with a clear golang interface. Implementations of e.g. CADF must be possible, but won't be discussed here.
+* dynamic loading of backends for new output formats are out of scope.
+
+## Use Cases
+
+1. As a cluster operator I want to enable audit logging of requests to the apiserver in order **to comply with given business regulations** regarding a subset of the 7 Ws of auditing:
+
+  - **what** happened?
+  - **when** did it happen?
+  - **who** initiated it?
+  - **on what** did it happen (e.g. pod foo/bar)?
+  - **where** was it observed (e.g. apiserver hostname)?
+  - from **where** was it initiated? (e.g. kubectl IP)
+  - to **where** was it going? (e.g. node 1.2.3.4 for kubectl proxy).
+
+1. Depending on the environment, as a cluster operator I want to **define the amount of audit logging**, balancing computational overhead for the apiserver with the detail and completeness of the log.
+
+1. As a cluster operator I want to **integrate with external systems**, which will have different requirements for the log format, network protocols and communication modes (e.g. pull vs. push).
+
+1. As a cluster operator I must be able to provide a **complete trace of changes to an object** to API objects.
+
+1. As a cluster operator I must be able to create a trace for **all accesses to a secret**.
+
+1. As a cluster operator I want to **define log-rotation** of a file-based output backend.
+
+1. As a cluster operator I must be able to log non-CRUD access like **kubectl exec**, when it started, when it finished and with which initial parameters.
+
```

Comment:
Add a use-case about possibility to record only filtered operations. See filters in following sections of this document.